### ISSUE TITLE
Add model search by "other"

### ISSUE THREAD

**Issue by fxmarty** [huggingface/hub-docs#148]
**Is your feature request related to a problem? Please describe.**
Models can not currently be searched by "other" without playing manually with the URL. See https://huggingface.co/models on the "Other" filter.

**Describe the solution you'd like**
Like for tasks where we have a clickable button "+16 tasks" that then allows to "Search tags", it would be neat to have in the "Other" category something like a "+3k tags", that then allows to search them as well. Could be useful as the number of models hosted on the Hub grows.

**Additional context**
![image](https://user-images.githubusercontent.com/9808326/168993159-41c38ccb-9d1a-4192-8ded-2a0ceb606283.png)

**Comment by osanseviero** [comment]
cc @gary149 @julien-c WDYT?

A challenge here is that there are thousands (and growing) number of other tags, so this could easily explode to being too many. I guess a search could work here, but the UX Might be strange.

Just FYI, the search experience is getting improved so this might be obsolete.

**Comment by julien-c** [comment]
yes i think we could do this at some point. Not super high prio though
